### ISSUE TITLE
Build failed if enum value is used in layout

### ISSUE THREAD

**Issue by m4tiz5zktonj** [markw65/monkeyc-optimizer#31]
Original code:
```monkey-c
import Toybox.Lang;
import Toybox.WatchUi;

class TestDrawable extends Drawable {
    enum TestEnum {
        VALUE,
    }

    public var value as TestEnum;

    function initialize(
        settings as
            {
                :identifier as Object,
                :locX as Numeric,
                :locY as Numeric,
                :width as Numeric,
                :height as Numeric,
                :visible as Boolean,
                :enumValue as TestEnum,
            }
    ) {
        Drawable.initialize(settings);

        value = settings.get(:enumValue) as TestEnum;
    }
}
```
```xml
<layout id="MainLayout">
    <drawable
        id="test"
        class="TestDrawable">
        <param name="locX">10</param>
        <param name="locY">10</param>
        <param name="width">10</param>
        <param name="height">10</param>
        <param name="enumValue">TestDrawable.VALUE</param>
    </drawable>
</layout>
```

Optimized:
```monkey-c
import Toybox.Lang;
import Toybox.WatchUi;

class TestDrawable extends Drawable {
    typedef TestEnum as Toybox.Lang.Number;

    public var value as TestEnum;

    function initialize(
        settings as
            {
                :identifier as Object,
                :locX as Numeric,
                :locY as Numeric,
                :width as Numeric,
                :height as Numeric,
                :visible as Boolean,
                :enumValue as TestEnum,
            }
    ) {
        Drawable.initialize(settings);
    }
}
```

Rez.mcgen:
```monkey-c
// Undefined symbol ':VALUE' detected.
var rez_cmp_local_custom_drawable_test = new TestDrawable({:identifier=>"test", :enumValue=>TestDrawable.VALUE, :width=>10, :locX=>10, :locY=>10, :height=>10});
```

**Comment by markw65** [comment]
> And the assignment value = settings.get(:enumValue) as TestEnum; is completely gone, even if I add System.println((findDrawableById("test") as TestDrawable).value); call to onUpdate!

This one turns out to be quite funny (well, it's funny to a compiler writer anyway).

The compiler has an optimization where if you assign a value to a variable that is already known to hold the same value, then it will kill the assignment.

The type `enum { VALUE }` can only hold one value (the number 0). So if you declare a variable with that type, the compiler "knows" that it holds that value. If you then assign an expression of that type to the variable, it knows that you didn't change the value, and removes the assignment.

The issue here is that a variable with a declared type is allowed to be null in the `initialize` method.

**Comment by markw65** [comment]
I guess I didn't add a test case. I'll make sure to do that this time...

**Comment by markw65** [comment]
Fixed by 348a48569a625c9caa8c15dee8fb76a1e8a616ed
